client/log_client.go

```diff
@@ -202,67 +204,94 @@ func (c *LogClient) getLatestRoot(ctx context.Context, trusted *types.LogRootV1)
 	return &logRoot, nil
 }
 
-// UpdateRoot retrieves the current SignedLogRoot, verifying it against roots this client has
-// seen in the past, and updating the currently trusted root if the new root verifies.
-func (c *LogClient) UpdateRoot(ctx context.Context) (*types.LogRootV1, error) {
+// GetRoot returns a copy of the latest trusted root.
+func (c *LogClient) GetRoot() *types.LogRootV1 {
+	c.rootLock.Lock()
+	defer c.rootLock.Unlock()
+
+	// Copy the internal trusted root in order to prevent clients from modifying it.
+	ret := c.root
+	return &ret
+}
+
+// updateRootIfNewer updates the current trusted root if the new root is newer
+// than the currently trusted root and its tree size is larger.
```

Comment:
Please add a big warning that this method should 
* Not be public
* Only be called from successful `getLatestRoot` which should really be called `getAndVerifyLatestRoot`

Reply:
Good point. I renamed `getLatestRoot` to `getAndVerifyLatestRoot`. For `updateRootIfNewer`, after looking more carefully at where it is used, I figured I could just get rid of the method.